### ISSUE TITLE
Support FOSS, choose a different Git forge

### ISSUE THREAD

**Issue by toastal** [Starz0r/AreWeAntiCheatYet#126]
GitHub is closed-source, proprietary and owned by Microsoft. This doesn't align with the core values of FOSS which is the what we want--better support for anti-cheat on FOSS platforms. Please consider moving to GitLab, Codeberg, Gitea, Sourcehut, or literally anything that's open source or open core.

**Comment by iSaluki** [comment]
GitHub is the biggest code collaboration platform around, and I doubt that this project will get moved over to GitLab. That said, I do have a mirror on GitLab, you can find it here: https://gitlab.com/iSaluki/AreWeAntiCheatYet

It is automatically synced with this repository, so if you don't want to use GitHub then you could use that for viewing the code, I don't have any push setup for it yet though.

**Comment by Starz0r** [comment]
While I understand the issues presented, there is simply more to it than just choose a “different” git forge.

GitHub is obviously the largest, therefor it brings a large community with it, and people are more likely to have an account on it than say GitLab.

Being closed-source does not really matter when it's a server hosted on a machine somewhere far out. This makes it entirely impossible to know if the code running on the server is the same code hosted in a public repository, so I see this as a non-issue. Although, I do understand the amount of proprietary JavaScript required to have a more functional experience on the site as a whole, but you'll have to bring that up with GitHub.

I also don't understand the hate that comes from Microsoft owning GitHub? This is such a pre-2008 topic to bring up, so I'd rather you leave it there. It has no place in the discussion, and you'd be better off leaving such hate in the past where it belongs.

As much as I hate contributing to monopolies, GitHub is rather benign, and provides a lot of benefits. Yes, a lot of the other forges do provide the same benefits, but they do not provide the same massive community and eyeballs I'd want for a project to be successful.

If there is some other reasoning that I'm not understanding or getting that's behind your post, then do reiterate and indulge me in it. Otherwise, this is the same reasoning I see everywhere else, and for me, it's not strong enough to move me off a platform when it's not mistreating me.

**Comment by IronMaltese** [comment]
This project should be mirrored over to Gitlab or Codeberg, that's what Libredirect did, and they respond to issue on both GitHub and Codeberg 
[This](https://github.com/libredirect/libredirect/issues/226) is the issue where this was all decided

**Comment by txtsd** [comment]
I kind of like Sourcehut as an alternative. The main advantage of being on GitHub is a large, already existing, user-base.

**Comment by Curve** [comment]
> The devs chose to use Github. If you disagree with them, make an alternative to AWACY on another hosting solution. Put your efforts where your opinion is instead of trying to evangelize.

Or just mirror it onto any platforms you deem good

**Comment by alexander-blackwell** [comment]
TLDR; Main a FOSS option, make GitHub the read-only mirror, and Lots of links to the main Git server for issues, etc.

I can see it both ways because GitHub is the biggest and most well-known; developers do themselves a disservice by not at the very least having their code viewable on this platform.

With that being said, GitHub could be the mirror while GitLab or, better yet, Gitea becomes the place where issues are tracked and written, and merge requests are made, with the GitHub mirror providing external links in many areas for those functions.

When a platform is willing to take down open source projects because other companies get hurt feelings when things like YouTube-DL exist, it's not a platform that I want to encourage the use of. While yes, they reversed course and the project is available again, they bent at the knee pretty quickly, so it will inevitably happen again.

**Comment by Starz0r** [comment]
A lot of people seem to think that this is an easy process or choice to make, when there are a lot of things to consider. Without getting too much into it, we currently use GitHub Issues to track major game changes or new problems with currently working games on the list. Everything is crowdsourced, which means getting users to have the lowest friction possible when submitting issues. I can assume for the most part that most GNU/Linux users have a GitHub account, and if not, creating one is very low effort. 

Besides that, we also utilize GitHub Actions and GitHub Pages, ruling out most competitors except for GitLabs, even which GitHub is more generous when it comes to GH Actions build minutes. A few other comments said we should use Sourcehut, but that is going to be a paid service in the future, and if I were some type of A-List software developer making tons of money, I wouldn't mind paying for it. However, I am not, and neither me, nor @Curve, are making any sort of money from this project. Assuming this is as much of a passion project as it is for him that it is for me, we don't make any monetary gain from this at all. The biggest cost for this project is currently the domain name, which is around 20USD yearly for me.

Yes, I don't always like the way Microsoft and GitHub run things, or the choices and decisions they make, but there is currently not a single competitor on the same level as GitHub. To be fair, none of these alternatives have to host us for free, nor do I expect them to. But it is extremely hard for me to make a decision like this when GitHub gives us so much for free. If we were somehow making money from donations, we'd be able to consider this better. We are making a net-zero from this project, and even if I did open up donations, I don't expect them to be able to cover the costs of operation forever.

If there was a way to allow users to participate without having an account at all, I could consider my options more closely, but I have no reason right now. At the current moment, I see no reason to stop using the platform, but I'll leave the idea open on the table for the future. Right now, there isn't anything explicit keeping me here, but there isn't anything pushing us away either.

I'll leave discussion open, but I'm closing this issue.
